ns for fx: change API to take nn.Module instead of GraphModule

vkuzo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #53828 ns for fx: move shadow lstm test to new API
* #53819 ns for fx: move shadow activations linear test to new API
* #53818 ns for fx: move shadow activations conv test to new API
* #53779 ns for fx: add support for lstm activation matching
* #53777 ns for fx: move linear activation test case to new API
* #53776 ns for fx: move compare activations for conv test to new API
* #53772 ns for fx: move lstm dynamic weight test case to new API
* #53765 ns for fx: move linear dynamic weight test case to new API
* #53764 ns for fx: move linear weight test case to new API
* #53761 ns for fx: move conv weight test case to new API
* #53748 ns for fx: extract common code in tests to util functions
* #53729 ns for fx: clean up API naming
* #53249 ns for fx: add support for logging inputs
* #53248 ns for fx: change dtype cast from once per N node to once per node
* **#53075 ns for fx: change API to take nn.Module instead of GraphModule**

Summary:

The input and output types should be `nn.Module`, to hide
the implementation detail that the pass is using FX.

Test Plan:

```
python test/test_quantization.py TestFXNumericSuiteCoreAPIs
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D26740548](https://our.internmc.facebook.com/intern/diff/D26740548)

vkuzo committed:
ns for fx: change API to take nn.Module instead of GraphModule

Summary:

The input and output types should be `nn.Module`, to hide
the implementation detail that the pass is using FX.

Test Plan:

```
python test/test_quantization.py TestFXNumericSuiteCoreAPIs
```

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

vkuzo committed:
Update on "ns for fx: change API to take nn.Module instead of GraphModule"


Summary:

The input and output types should be `nn.Module`, to hide
the implementation detail that the pass is using FX.

Test Plan:

```
python test/test_quantization.py TestFXNumericSuiteCoreAPIs
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D26740548](https://our.internmc.facebook.com/intern/diff/D26740548)

[ghstack-poisoned]

vkuzo committed:
Update on "ns for fx: change API to take nn.Module instead of GraphModule"


Summary:

The input and output types should be `nn.Module`, to hide
the implementation detail that the pass is using FX.

Test Plan:

```
python test/test_quantization.py TestFXNumericSuiteCoreAPIs
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D26740548](https://our.internmc.facebook.com/intern/diff/D26740548)

[ghstack-poisoned]

vkuzo committed:
Update on "ns for fx: change API to take nn.Module instead of GraphModule"


Summary:

The input and output types should be `nn.Module`, to hide
the implementation detail that the pass is using FX.

Test Plan:

```
python test/test_quantization.py TestFXNumericSuiteCoreAPIs
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D26740548](https://our.internmc.facebook.com/intern/diff/D26740548)

[ghstack-poisoned]

jerryzh168 commented on torch/quantization/ns/numeric_suite_core_apis_fx.py:
```diff
@@ -241,62 +280,68 @@ def add_activation_info_to_dict(
+    Same thing as prepare_model_outputs, but for an `a_shadows_b` model.
+    TODO(future PR): real docblock
+    """
+    tracer_a, tracer_b = NSTracer(), NSTracer()
```
you can probably just use the same tracer I think

vkuzo committed:
Update on "ns for fx: change API to take nn.Module instead of GraphModule"


Summary:

The input and output types should be `nn.Module`, to hide
the implementation detail that the pass is using FX.

Test Plan:

```
python test/test_quantization.py TestFXNumericSuiteCoreAPIs
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D26740548](https://our.internmc.facebook.com/intern/diff/D26740548)

[ghstack-poisoned]
